File: backend/src/main/java/com/digitaltwin/platform/service/NotificationService.java
package com.digitaltwin.platform.service;

import com.digitaltwin.platform.dto.notification.NotificationResponse;

import java.util.List;
import java.util.Map;

/**
 * Manages in-app notifications for a user: creation (used internally
 * by other services, e.g. "your resume analysis is ready"), listing,
 * and marking read.
 */
public interface NotificationService {

    NotificationResponse create(Long userId, String title, String message, String type);

    List<NotificationResponse> getMyNotifications(Long userId, boolean unreadOnly);

    Map<String, Long> getUnreadCount(Long userId);

    NotificationResponse markAsRead(Long userId, Long notificationId);

    void markAllAsRead(Long userId);

    void delete(Long userId, Long notificationId);
}

File: frontend/src/services/notificationService.js
import apiClient from './apiClient.js';

const notificationService = {
  getMyNotifications: (unreadOnly = false) =>
    apiClient.get('/notifications', { params: { unreadOnly } }).then((r) => r.data),
  getUnreadCount: () => apiClient.get('/notifications/unread-count').then((r) => r.data),
  markAsRead: (id) => apiClient.patch(`/notifications/${id}/read`).then((r) => r.data),
  markAllAsRead: () => apiClient.patch('/notifications/read-all').then((r) => r.data),
  delete: (id) => apiClient.delete(`/notifications/${id}`).then((r) => r.data),
};

export default notificationService;
